pulp_ansible/tests/conftest.py

```diff
@@ -0,0 +1,80 @@
+import uuid
+
+import pytest
+
+from pulp_smash.pulp3.utils import gen_distribution, gen_repo
+
+from pulpcore.client.pulp_ansible import (
+    ContentCollectionVersionsApi,
+    DistributionsAnsibleApi,
+    RepositoriesAnsibleApi,
+    RemotesCollectionApi,
+)
+
+from pulp_ansible.tests.functional.utils import (
+    gen_ansible_client,
+)
+
+
+# Bindings API Fixtures
+
+
+@pytest.fixture(scope="session")
+def ansible_bindings_client():
+    """Provides the Ansible bindings client object."""
+    return gen_ansible_client()
+
+
+@pytest.fixture(scope="session")
+def ansible_repo_api_client(ansible_bindings_client):
+    """Provides the Ansible Repository API client object."""
+    return RepositoriesAnsibleApi(ansible_bindings_client)
+
+
+@pytest.fixture(scope="session")
+def ansible_distro_api_client(ansible_bindings_client):
+    """Provides the Ansible Distribution API client object."""
+    return DistributionsAnsibleApi(ansible_bindings_client)
+
+
+@pytest.fixture(scope="session")
+def ansible_collection_version_api_client(ansible_bindings_client):
+    """Provides the Ansible Content Collection Version API client object."""
+    return ContentCollectionVersionsApi(ansible_bindings_client)
+
+
+@pytest.fixture(scope="session")
+def ansible_remote_collection_api_client(ansible_bindings_client):
+    """Provides the Ansible Collection Remotes API client object."""
+    return RemotesCollectionApi(ansible_bindings_client)
+
+
+# Object Generation Fixtures
+
+
+@pytest.fixture
+def ansible_repo(ansible_repo_api_client, gen_object_with_cleanup):
+    """Creates an Ansible Repository and deletes it at test cleanup time."""
+    return gen_object_with_cleanup(ansible_repo_api_client, gen_repo())
```

Comment:
Maybe this should be a factory too.

Reply:
I suspect over time end up with both versions, but this is the kind I need now. I like having both options because if need exactly one you don't have to call the factory in your test itself. This is debatable though, more fixtures for less lines of test code...

Reply:
Yeah, it's a matter of building habits. So no require multiple repos here.